components/allocator/lib.rs

```diff
@@ -16,19 +15,90 @@ pub use platform::*;
 
 #[cfg(all(feature = "unstable", not(windows)))]
 mod platform {
-    extern crate jemallocator;
+    extern crate jemalloc_sys as ffi;
 
-    pub use self::jemallocator::Jemalloc as Allocator;
-    use std::os::raw::c_void;
+    use std::alloc::{GlobalAlloc, Layout, Opaque, System};
+    use std::os::raw::{c_int, c_void};
 
     /// Get the size of a heap block.
     pub unsafe extern "C" fn usable_size(ptr: *const c_void) -> usize {
-        jemallocator::usable_size(ptr)
+        ffi::malloc_usable_size(ptr as *const _)
     }
 
     /// Memory allocation APIs compatible with libc
     pub mod libc_compat {
-        pub use super::jemallocator::ffi::{malloc, realloc, free};
+        pub use super::ffi::{malloc, realloc, free};
+    }
+
+    pub struct Allocator;
```

Comment:
Could we get an issue on file to undo this?

Reply:
https://github.com/servo/servo/issues/20645